src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let resultObject = { ...state };
+  const resultArray = [];
+  const ADD = 'addProperties';
+  const REMOVE = 'removeProperties';
+  const CLEAR = 'clear';
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ADD: {
+        resultArray.push(Object.assign({}, resultObject, action.extraData));
+        Object.assign(resultObject, action.extraData);
+        break;
+      }
+
+      case REMOVE: {
+        for (const key of action.keysToRemove) {
+          if (resultObject.hasOwnProperty(key)) {
+            delete resultObject[key];
+          }
+        }
+        resultArray.push({ ...resultObject });
```

Comment:
move it in end of func, because you use it in eny cases, just make manipulation with result obj and in the end make a copy and push

Reply:
done